### DISCUSSION TITLE
don't know why it still feels slower than fzf.vim

### DISCUSSION THREAD

**Discussion by ssh352** [ibhagwan/fzf-lua#1190] [Q&A] [answered]
```
        config = function()
            -- calling `setup` is optional for customization
            require("fzf-lua").setup({
                -- "fzf-vim",
                -- "fzf-native",
                "max-perf",
                winopts = { preview = { hidden = "nohidden" } },
                files = {
                    -- Use FZF_DEFAULT_COMMAND to keep the default behavior.
                    cmd = vim.env.FZF_DEFAULT_COMMAND,
                },
                -- default_previewer = "bat_native",
                default_previewer = "bat",
            })
        end,
        keys = {
            {
                "<leader>f",
                -- ":FzfLua files<CR>",
                -- ":Files<CR>",
                function()
                    require("fzf-lua").files()
                end,
                desc = "Find Files from CWD",
                silent = true,
            },
```

```
export FZF_DEFAULT_COMMAND='rg --files --no-hidden --follow 2> /dev/null' # --no-hidden do not search hidden files or dirs
```

I'd really like to make it as fast as fzf.vim, thanks!

**Comment by ibhagwan** [comment]
I apologize for what may seem as an aggressive response but it’s very hard for me to deal with “feelings”, what exactly is slower? Opening/ closing? Enumerating the file system? Typing? Speed of preview? Have you benchmarked and can tell it is slower or just a “feeling”?

As long as we remain the realm of “feelings” I can’t really help you, if you are truly seeking help in this regard do some work and provide meaningful details before opening a discussion.

**Reply by ssh352** [reply]
excuse my question. I just eyeballed the file list show up many times an I am pretty sure in fzf.vim it shows up faster. Please advise how do I prove it ?

**Reply by ibhagwan** [reply]
I still don’t understand what this means, which part feels slow? From the time you press the key bind till the interface shows up? Till the list is full of files? Is it a big project? Small?

Or are you just spamming your key bind and it “feels” slower to you?

you’re not making it easy to help you.

**Comment by ibhagwan** [comment]
Here’s an example of a productive “slower than fzf.vim” discussion: https://github.com/ibhagwan/fzf-lua/issues/904

If this applies to you (the internal mapping of `Esc<->Ctrl-C`), you can do what’s described in https://github.com/ibhagwan/fzf-lua/issues/904#issuecomment-1761013882 (at the risk of the occasional hang when closing with ctrl-c due to #1091, https://github.com/neovim/neovim/issues/20726):
```lua
  vim.api.nvim_create_autocmd("FileType",
    {
      group = vim.api.nvim_create_augroup("FzfLuaCtrlC", { clear = true }),
      pattern = "fzf",
      callback = function(e)
        vim.keymap.set("t", "<C-c>", "<C-c>", { buffer = e.buf, silent = true })
      end,
    })
```

**Reply by ssh352** [reply]
I've made a screen recording to demonstrate the difference: the first is fzf.vim, the second is fzf-lua with config posted above. 

https://github.com/ibhagwan/fzf-lua/assets/41779116/c159e999-4bc2-4fb7-93a1-019b3d5e0660

**Reply by ibhagwan** [reply]
Quite honestly I don’t see a big difference, nor will I know how to “fix” this, if this truly bothers you the only advice I can give you is to stick with `fzf.vim`.

**Reply by ssh352** [reply]
apologize, I've figured out, I have \f \fe \fE all mapped. It seems after I remove \fe \fE mapping, \f is as fast as fzf.vim :Files. I don't understand why though..

**Reply by ibhagwan** [reply, marked as answer]
Once you have a mapping for both `\f` and `\fE` (or any other letter after `f`) neovim has to wait in “operator pending” mode in order to give you an option to press the next letter, when you remove the `\fE` mapping neovim knows not to expect any other keypress after `\f` and thus executes it immediately.

**Reply by ssh352** [reply]
Thanks it is nice to affirm fzf-lua is as fast as fzf.vim!

**Reply by ibhagwan** [reply]
FYI, until your last comment (even after the screenshot) I had no idea that you meant a delay since you press your keybind until the interface starts to appear, it would’ve been very helpful to explain that and could’ve saved both of us some time.

All this time I was thinking that you ran `:Files` and `:FzfLua files` and are comparing the speeds, not that **you have a delay in your key bind**.

Anyways, I’m glad you got this resolved.
